### PR TITLE
Compiler Expiration: clear on reload

### PR DESCRIPTION
**Pull request by danielbeardsley**
Fix a bug where we wouldn't clear the expire timeout when replacing
the webpack compiler. Thus if you reloaded, {timeout} seconds later
it would kill the compiler, seemingly out of no-where.

Now we clear the timeout before replacing (in set) and before
removing it.

### CODE REVIEW

**Comment by jarstelfox** [comment]
CR 🌵

File: compiler-collection.js [hunk 1]
```diff
@@ -7,25 +7,33 @@ module.exports = (expireAfterSeconds) => {
     if (!compiler || !expireMs) {
        return;
     }
-    clearTimeout(compiler.expiredTimeout);
```
Comment (danielbeardsley) [inline compiler-collection.js:10]: I don't like the way this is done...  the timer is attached to the compiler, but it doesn't kill *that*  compiler, it just kills the compiler currently store for the repo. If we just had the timeout kill the compiler its attached to, there'd be less to worry about.

**Comment by danielbeardsley** [comment]
dev_block 👍  on my own suggestion.

File: compiler-collection.js (hunk 1 above)
Comment (danielbeardsley) [reply compiler-collection.js:10]: Fixed!

**Comment by danielbeardsley** [comment]
un_dev_block 👍

**Comment by jarstelfox** [comment]
CR 🌵

**Comment by sterlinghirsh** [comment]
CR :zap:

**Pull request merged**